app/code/Magento/Catalog/Block/Product/ImageFactory.php

```diff
@@ -0,0 +1,179 @@
+<?php
```

Comment:
It's not a good idea to port the whole class and use only one method. Please, consider to provide different implementation compatible with 2.2-develop.

Reply:
Disagree with that. @sidolov why we need to reinvent the wheel?

Reply:
I'm not sure if this new class is actually used though.

Reply:
@orlangur I agree that we shouldn't implement completely new solution, but introducing code that never used not a good idea. We should port only the fragments that resolves the problem

Reply:
@sidolov but it requires more efforts and make codebase 2.2/2.3 less consistent. I see your point but as for me some unused code is lesser evil.

Reply:
@orlangur it's pretty simple, for 2.2 we need just one method from 2.3 version. We trying to introduce new code that will be marked as unused and deprecated